datamad2/models/grants.py

```diff
@@ -70,6 +70,26 @@ class Grant(models.Model):
     dmp_agreed = models.BooleanField(default=False)
     dmp_agreed_date = MonitorField(monitor='dmp_agreed', when=[True])
 
+    @property
+    def update_linked_grants(self):
+        editable_grant_info = {"alt_data_contact": self.alt_data_contact, 
+        "alt_data_contact_email": self.alt_data_contact_email, 
+        "alt_data_contact_phone": self.alt_data_contact_phone, 
+        "other_data_centre": self.other_data_centre, 
+        "date_contacted_pi": self.date_contacted_pi, 
+        "will_grant_produce_data": self.will_grant_produce_data, 
+        "datasets_delivered": self.datasets_delivered, 
+        "sanctions_recommended": self.sanctions_recommended, 
+        "dmp_agreed": self.dmp_agreed}
+
+        if self.parent_grant:
+            Grant.objects.filter(pk=self.parent_grant.pk).update(**editable_grant_info)
+
+        child_grants = self.child_grant.get_queryset()
+        for child_grant in child_grants:
+            Grant.objects.filter(pk=child_grant.pk).update(**editable_grant_info)
```

Comment:
Yes, you would only expect one result from `Grant.objects.filter(pk=self.parent_grant.pk).update(**editable_grant_info)`, but it has to use filter rather than update otherwise there is an error message: `Grant object has no attribute 'update'`

I also wasn't updating siblings.

Removed the else, thought there might be a chance that a grant is a child and a parent?

So have updated to:


```
        if self.parent_grant:
            # if there is a parent grant, I am a child grant. Update parent.
            Grant.objects.filter(pk=self.parent_grant.pk).update(**editable_grant_info)
            
            # Get and update siblings
            Grant.objects.filter(parent_grant=self.parent_grant).update(**editable_grant_info)

        # I might be a parent grant - get any child grants and update
        child_grants = self.child_grant.get_queryset()
        child_grants.update(**editable_grant_info)
```